Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -919,6 +920,28 @@ static inline bool expectsLegacyImplicitRubberBandControl()
 
 #if ENABLE(WRITING_TOOLS)
 
+bool WebPageProxy::shouldEnableWritingToolsRequestedTool(WebCore::WritingTools::RequestedTool tool) const
+{
+    WTRequestedTool requestedTool = convertToPlatformRequestedTool(tool);
+
+    if (requestedTool == WTRequestedToolIndex)
+        return true;
+
+    bool editorStateIsContentEditable = false;
+    bool allowProofreadingAndRewritingTools = true;
+
+    auto& editorState = this->editorState();
+    editorStateIsContentEditable = editorState.isContentEditable;
+
+    if (editorStateIsContentEditable)
+        allowProofreadingAndRewritingTools = editorState.hasPostLayoutData() && !editorState.postLayoutData->paragraphContextForCandidateRequest.isEmpty();
+
+    if (requestedTool == WTRequestedToolCompose)
```

Comment:
this if can be moved above to the preceding if to possibly avoid an extra if

Reply:
Done. And also got rid of the `allowProofreadingAndRewritingTools` variable.